tools/hsmtool.c

```diff
@@ -368,6 +371,55 @@ static int guess_to_remote(const char *address, struct node_id *node_id,
 	return 1;
 }
 
+static int dumponchaindescriptors(const char *hsm_secret_path, const char *passwd)
+{
+	struct secret hsm_secret;
+	u8 bip32_seed[BIP32_ENTROPY_LEN_256];
+	u32 salt = 0;
+	struct ext_key master_extkey;
+	char *enc_xpub, *descriptor, checksum[DESCRIPTOR_CHECKSUM_LENGTH + 1];
+
+	if (passwd)
+		get_encrypted_hsm_secret(&hsm_secret, hsm_secret_path, passwd);
+	else
+		get_hsm_secret(&hsm_secret, hsm_secret_path);
+
+	/* We use m/0/0/k as the derivation tree for onchain funds. */
+
+	/* The root seed is derived from hsm_secret using hkdf.. */
+	do {
+		hkdf_sha256(bip32_seed, sizeof(bip32_seed),
```

Comment:
Duplicating this code from `hsmd/hsmd.c#populate_secretstuff` is a bad idea. Needs to be a shared routine that both hsmd and here use.

Reply:
That's how we've created tool since now ?..

Reply:
I don't follow. We really want this code to be shared so that if one changes, the other also gets updated.